### PR TITLE
fix: LGTM code quality and Added docs

### PR DESCRIPTION
**Pull request by ayaankhan98**
#### Description of Change
- fixed code quality - #880 
- added documentation

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
Author name in documentaion is missiing, i don't know who has written this file initially as i had not tracked the commit history.

### CODE REVIEW

**Comment by Panquesito7** [comment]
> Author name in documentation is missing

Name is `leoyang429`, PR #704.

**Comment by kvedala** [comment]
@ayaankhan98 you probably have to rebase your fork with the master. Thats why there are so many additional commits

**Comment by ayaankhan98** [comment]
> @ayaankhan98 you probably have to rebase your fork with the master. Thats why there are so many additional commits

okay i will do that for sure

**Comment by ayaankhan98** [comment]
@kvedala a new file is pushed review this one too, after this we can merge this

File: sorting/comb_sort.cpp [hunk 1]
```diff
@@ -1,49 +1,82 @@
...
-    CombSort(a, 1, n);
-
-    for (int i = 1; i <= n; ++i) std::cout << a[i] << ' ';
-    return 0;
+  int n;
+  std::cin >> n;
+  int *arr = new int[n];
+  for (int i = 0; i < n; ++i) std::cin >> arr[i];
+  CombSort(arr, 0, n);
+  for (int i = 0; i < n; ++i) std::cout << arr[i] << ' ';
```
Comment (kvedala) [inline sorting/comb_sort.cpp]: 👍 can you also add the self test check as well using the `std::is_sorted`?

Comment (ayaankhan98) [reply sorting/comb_sort.cpp]: yeah sure why not

File: data_structures/binaryheap.cpp [hunk 2]
```diff
@@ -13,10 +13,14 @@ class MinHeap {
...
      */
-    MinHeap(int capacity);
+    explicit MinHeap(int cap) {
+        heap_size = 0;
+        capacity = cap;
+        harr = new int[cap];
+    }
 
     /** to heapify a subtree with the root at given index
      */
```
Comment (kvedala) [inline data_structures/binaryheap.cpp:26]: please delete this line

Comment (ayaankhan98) [reply data_structures/binaryheap.cpp:26]: removed

Comment (kvedala) [reply data_structures/binaryheap.cpp:26]: > removed

Oh, my bad. I misread the function name and thought it was the same as the constructor. The commit 48a163e51ab53a73c9edbb02633e6a3673865cea is the last good commit. Please revert to it. Thank you

Comment (ayaankhan98) [reply data_structures/binaryheap.cpp]: no problem, i too did not looked closely, i thought you were right

**Comment by mishraabhinn** [comment]
You could  also included a merge function to merge two disjoint set.

**Comment by ayaankhan98** [comment]
> You could also included a merge function to merge two disjoint set.

thanks for suggestion @mishraabhinn actually i had not written this code i just picked up this file to fix the LGTM code quality issue suddenly i realize that documentaion is also poor so i fixed that too, but this merge function doesn't strike to me at that time.
BTW i will try to implement this is another PR soon.

**Comment by kvedala** [comment]
> You could also included a merge function to merge two disjoint set.



> > You could also included a merge function to merge two disjoint set.
> 
> thanks for suggestion @mishraabhinn actually i had not written this code i just picked up this file to fix the LGTM code quality issue suddenly i realize that documentaion is also poor so i fixed that too, but this merge function doesn't strike to me at that time.
> BTW i will try to implement this is another PR soon.

👍  @mishraabhinn please feel free to contribute your suggested algorithm

**Comment by mishraabhinn** [comment]
> > You could also included a merge function to merge two disjoint set.
> 
> > > You could also included a merge function to merge two disjoint set.
> > 
> > 
> > thanks for suggestion @mishraabhinn actually i had not written this code i just picked up this file to fix the LGTM code quality issue suddenly i realize that documentaion is also poor so i fixed that too, but this merge function doesn't strike to me at that time.
> > BTW i will try to implement this is another PR soon.
> 
> 👍 @mishraabhinn please feel free to contribute your suggested algorithm

I got little confused over union and merge. 

merge(a,b)
{
a=find(a);
b=find(b);

if(a==b)
return;
else
parent[a]=b;
	
}



`

**Comment by ayaankhan98** [comment]
> > > You could also included a merge function to merge two disjoint set.
> > 
> > 
> > > > You could also included a merge function to merge two disjoint set.
> > > 
> > > 
> > > thanks for suggestion @mishraabhinn actually i had not written this code i just picked up this file to fix the LGTM code quality issue suddenly i realize that documentaion is also poor so i fixed that too, but this merge function doesn't strike to me at that time.
> > > BTW i will try to implement this is another PR soon.
> > 
> > 
> > +1 @mishraabhinn please feel free to contribute your suggested algorithm
> 
> I got little confused over union and merge.
> 
> merge(a,b)
> {
> a=find(a);
> b=find(b);
> 
> if(a==b)
> return;
> else
> parent[a]=b;
> 
> }
> 
> `

as far as i know both merge and union operations are same.

**Comment by mishraabhinn** [comment]
> > > > You could also included a merge function to merge two disjoint set.
> > > 
> > > 
> > > > > You could also included a merge function to merge two disjoint set.
> > > > 
> > > > 
> > > > thanks for suggestion @mishraabhinn actually i had not written this code i just picked up this file to fix the LGTM code quality issue suddenly i realize that documentaion is also poor so i fixed that too, but this merge function doesn't strike to me at that time.
> > > > BTW i will try to implement this is another PR soon.
> > > 
> > > 
> > > +1 @mishraabhinn please feel free to contribute your suggested algorithm
> > 
> > 
> > I got little confused over union and merge.
> > merge(a,b)
> > {
> > a=find(a);
> > b=find(b);
> > if(a==b)
> > return;
> > else
> > parent[a]=b;
> > }
> > `
> 
> as far as i know both merge and union operations are same.

Sorry for this comment on the pr. Actually i didn't read the file properly due less comments and documentation. So i impulsively commented.

**Pull request merged**